dpdata/abacus/scf.py

```
@@ -759,7 +759,15 @@
 
     # ATOMIC_SPECIES block
     out = "ATOMIC_SPECIES\n"
-    ppfiles = process_file_input(ndarray2list(pp_file), data["atom_names"], "pp_file")
+    if pp_file is not None:
+        ppfiles = process_file_input(
+            ndarray2list(pp_file), data["atom_names"], "pp_file"
+        )
+    else:
+        warnings.warn(
```

Ruff

No explicit `stacklevel` keyword argument found

(B028)